### PR TITLE
SAMOA-72 New Boosting algorithm for VHT

### PR DESCRIPTION
**Pull request by fobeligi**
This pull request provides BoostVHT algorithm, an online boosting algorithm that leverages VHT algorithm as a base learner. BoostVHT design allows the separation of the model boosting from its training, thus obtaining order of magnitude speedups compared to the state-of-the-art.

Since a number of classes that are used by VHT algorithm are changed, we performed tests to assure that VHT's performance was not affected, before this pull request.

Below I refer briefly to a number of changes:
* `DecisionStump` class was initially imported from MOA in order to perform some sanity checks. We kept the code in this PR.
* In `AttributeSplitSuggestion` class we change the public constructor to set `merit = Double.NEGATIVE_INFINITY;`, instead of doing nothing, which didn't affect VHT's performance in tests.
* Methods in `static class Builder` of `ModelAggregatorProcessor` are changed to return a generic, as we extend the `ModelAggregatorProcessor.Builder` class in `BoostMAProcessor extends ModelAggregatorProcessor` class.
* `LocalStatisticsProcessor` class was changed to also handle an `AttributeSliceEvent`, an event that was introduced for BoostVHT algorithm.
* Classes that are used both from VHT and BoostVHT algorithms have a number of access level changes.

Sure there are more changes than the ones I referred to above. 
Αny feedback would be appreciated.

### CODE REVIEW

**Comment by thvasilo** [comment]
Seems like there's some configuration error for Travis, causing the builds to fail.

**Comment by nicolas-kourtellis** [comment]
Hi, any updates with this PR?
@fobeligi did you manage to build it successfully?

**Comment by thvasilo** [comment]
Hello @nicolas-kourtellis, IIRC the PR builds fine locally, the failure comes from a misconfigured Travis.

File: samoa-api/src/main/java/org/apache/samoa/learners/classifiers/trees/LocalStatisticsProcessor.java [hunk 1]
```diff
@@ -180,16 +208,16 @@ public Processor newProcessor(Processor p) {
...
   }
 
   private AttributeClassObserver newNominalClassObserver() {
-    return (AttributeClassObserver) this.nominalClassObserver.copy();
+    return new NominalAttributeClassObserver(); //further investigate this change
   }
 
   private AttributeClassObserver newNumericClassObserver() {
-    return (AttributeClassObserver) this.numericClassObserver.copy();
+    return new GaussianNumericAttributeClassObserver();//further investigate this change
```
Comment (csterling) [inline samoa-api/src/main/java/org/apache/samoa/learners/classifiers/trees/LocalStatisticsProcessor.java]: @fobeligi Can you please comment on why this change was necessary, and the implications of it?

**Comment by csterling** [comment]
After further discussion with @abifet these changes have been deemed acceptable. +1

**Comment by gdfm** [comment]
Thanks for integrating this!
